src/plugins/intel_npu/src/plugin/src/compiled_model.cpp

```diff
@@ -187,16 +187,20 @@ ov::Any CompiledModel::get_property(const std::string& name) const {
     OPENVINO_THROW("Unsupported property ", name);
 }
 
-const std::shared_ptr<const NetworkDescription>& CompiledModel::get_network_description() const {
-    return _networkPtr;
+const NetworkDescription& CompiledModel::get_network_description() const {
+    return *_networkPtr;
 }
 
 const Config& CompiledModel::get_config() const {
     return _config;
 }
 
-const ov::SoPtr<ICompiler>& CompiledModel::get_compiler() const {
-    return _compiler;
+const ICompiler& CompiledModel::get_compiler() const {
```

Comment:
High chances there is a reason in the code history for existence of this SO pointer.

Reply:
Maybe, but didn't find the reason. If you know the reason, please let us know.

Reply:
Refer first to ticket #111849 which introduced the change.
Or perhaps @lmielick could give the answer straight away.

Was there a reason for switching to return a `ov::SoPtr<ICompiler>&`? This PR tempts to move things backwards.

Reply:
Yes, there is a strong reason we use a SoPtr here. In CiP case the compiler is loaded a DLL (see compiler.cpp).
SoPtr stands for Shared Object Pointer, and is used to keep track of references the the DLL along related object shared_ptr. DLL will be unloaded when last SoPtr is destroyed.

Reply:
I'm trying to understand where this method is used and need to increment the reference but didn't find yet. Could you please point me to the place where this is needed in this way?

Reply:
I guess removing this will not crash, because we keep compiler reference anyways (in Plugin object or somewhere), but it's not great to have non-local assumptions on code correctness.

One flow I'm aware of would be the profiling case where we need to call the compiler after inference, so we cannot unload the compiler DLL even if we decided that's OK otherwise.

Side-note: It's not a issue in here, but it's worth noting that is is impossible to transfer shared_ptr instancess across DLL boundary, as shared_ptr involves a virtual dtor. So effectively DLL cannot be unloaded as long as any shared_ptr created inside DLL is held.